daft/dataframe/dataframe.py

```diff
@@ -65,7 +65,7 @@ def __init__(self, plan: logical_plan.LogicalPlan) -> None:
         Args:
             plan: LogicalPlan describing the steps required to arrive at this DataFrame
         """
-        if not isinstance(plan, logical_plan.LogicalPlan):
+        if not isinstance(plan, (logical_plan.LogicalPlan, rust_logical_plan.RustLogicalPlanBuilder)):
```

Comment:
Should we change the type annotation of the `plan` arg and the `_plan` property accessor to be a union of `LogicalPlan` and `RustLogicalPlanBuilder`?